src/psyclone/dynamo0p3.py

```diff
@@ -9759,7 +9761,23 @@ def psyir_expression(self):
 
         if self.is_literal:
             reader = FortranReader()
-            return reader.psyir_from_expression(self.name, symbol_table)
+            if self.precision:
+                # Ensure any associated precision symbol is in the table.
+                symbol_table.add_lfric_precision_symbol(self.precision)
+            try:
+                lit = reader.psyir_from_expression(self.name, symbol_table)
+            except SymbolError as err:
+                raise InternalError(
+                    f"Unexpected literal expression '{self.name}' when "
+                    f"processing kernel '{self.call.name}'.") from err
+
+            # Sanity check that the resulting expression is a literal.
+            if lit.walk(Reference):
```

Comment:
Couldn't this just be `not isinstance(lit, Literal)`?

Reply:
I have answered my own question once I saw the test, so please ignore.

Reply:
I'd love to say I thought of this ahead of time but what actually happened is I ran the tests and some failed because of this :-)